cmd/serve.go

```diff
@@ -185,6 +186,14 @@ func getUpstreamCaches(ctx context.Context, cmd *cli.Command) ([]upstream.Cache,
 			return nil, fmt.Errorf("error parsing --upstream-cache=%q: %w", us, err)
 		}
 
+		var priority int64 = -1
```

Comment:
Is zero a valid priority? I feel like 0 is invalid and thus you should just start with zero and send uint64 everywhere. If not then fair enough.